x-pack/plugins/security/server/spaces/secure_spaces_client_wrapper.ts

```diff
@@ -265,6 +269,30 @@ export class SecureSpacesClientWrapper implements ISpacesClient {
       }
     }
 
+    // Fetch saved objects to be removed for audit logging
+    if (this.auditLogger.enabled) {
+      const finder = this.spacesClient.createSavedObjectFinder(id);
+      try {
+        for await (const response of finder.find()) {
+          response.saved_objects.forEach((savedObject) => {
+            const isOnlySpace = !savedObject.namespaces || savedObject.namespaces.length === 1;
```

Comment:
Sorry, I just noticed a few things that I should have seen earlier:

1. You don't need to worry about handling `namespaces: undefined`, it's not a realistic scenario. 
   a. If the object type is share-capable (`namespaceType: 'multiple-isolated'`) or shareable (`namespaceType: 'multiple'`) *AND* the underlying document has `namespaces: undefined` then it won't turn up in the search results.
   b. If, on the other hand, the underlying document is a single-namespace saved object type (`namespaceType: 'isolated'`), the underlying document should always have `namespaces: undefined`, but [before the SOR returns the resulting saved object it adds `namespaces: ['currentSpace']`](https://github.com/elastic/kibana/blob/ca05637afa9fc822d0e7abd6f98c6370d2327656/src/core/server/saved_objects/service/lib/repository.ts#L2182-L2184). This is so that consumers can rely on a single field to check what space(s) an object exists in.
2. This implementation will treat an object with `namespaces: ['*']` as being deleted, when in reality that object won't be deleted, it won't be changed at all.
3. There is also an interesting corner case where an object could have `namespaces: ['*', 'spaceThatIsBeingDeleted']`; this is semantically almost identical to `namespaces: ['*']`, but the object is still being changed, so in that case so we should still create an audit record for it.

So I think you need to make this change along with updated tests:
```diff
diff --git a/x-pack/plugins/security/server/spaces/secure_spaces_client_wrapper.ts b/x-pack/plugins/security/server/spaces/secure_spaces_client_wrapper.ts
index 52fc1ea84a6..c4321664320 100644
--- a/x-pack/plugins/security/server/spaces/secure_spaces_client_wrapper.ts
+++ b/x-pack/plugins/security/server/spaces/secure_spaces_client_wrapper.ts
@@ -17,6 +17,7 @@ import type {
   LegacyUrlAliasTarget,
   Space,
 } from '../../../spaces/server';
+import { ALL_SPACES_ID } from '../../common/constants';
 import type { AuditLogger } from '../audit';
 import { SavedObjectAction, savedObjectEvent, SpaceAuditAction, spaceAuditEvent } from '../audit';
 import type { AuthorizationServiceSetup } from '../authorization';
@@ -275,7 +276,12 @@ export class SecureSpacesClientWrapper implements ISpacesClient {
       try {
         for await (const response of finder.find()) {
           response.saved_objects.forEach((savedObject) => {
-            const isOnlySpace = !savedObject.namespaces || savedObject.namespaces.length === 1;
+            const { namespaces = [] } = savedObject;
+            const isOnlySpace = namespaces.length === 1; // We can always rely on the `namespaces` field having >=1 element
+            if (namespaces.includes(ALL_SPACES_ID) && !namespaces.includes(id)) {
+              // This object exists in All Spaces and its `namespaces` field isn't going to change; there's nothing to audit
+              return;
+            }
             this.auditLogger.log(
               savedObjectEvent({
                 action: isOnlySpace
```